File: _work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/process.py
from __future__ import annotations

from .client import HttpClient
from .types import ProcessInfo, ProcessTopInfo


class ProcessManager:
    def __init__(self, client: HttpClient, sandbox_id: str):
        self._client = client
        self._sandbox_id = sandbox_id

    async def list(self) -> dict:
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/processes"
        )

    async def kill(
        self, pid: int, signal: str | None = None
    ) -> dict:
        body: dict = {"pid": pid}
        if signal is not None:
            body["signal"] = signal
        return await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/processes/kill", body
        )

    async def top(self) -> dict:
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/processes/top"
        )

File: _work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk/src/process.ts
import type { HttpClient } from "./client.js";

export interface ProcessInfo {
  pid: number;
  user: string;
  cpu: string;
  memory: string;
  command: string;
}

export interface ProcessTopInfo {
  pid: number;
  cpu: string;
  mem: string;
  vsz: number;
  rss: number;
  command: string;
}

export class ProcessManager {
  constructor(private client: HttpClient, private sandboxId: string) {}

  async list(): Promise<{ processes: ProcessInfo[] }> {
    return this.client.get(`/sandbox/sandboxes/${this.sandboxId}/processes`);
  }

  async kill(pid: number, signal?: string): Promise<{ killed: number; signal: string }> {
    return this.client.post(`/sandbox/sandboxes/${this.sandboxId}/processes/kill`, { pid, signal });
  }

  async top(): Promise<{ processes: ProcessTopInfo[] }> {
    return this.client.get(`/sandbox/sandboxes/${this.sandboxId}/processes/top`);
  }
}
